src/qt/bitcoin.cpp

```diff
@@ -510,7 +510,12 @@ int GuiMain(int argc, char* argv[])
     // Show help message immediately after parsing command-line options (for "-lang") and setting locale,
     // but before showing splash screen.
     if (HelpRequested(gArgs) || gArgs.IsArgSet("-version")) {
-        HelpMessageDialog help(nullptr, gArgs.IsArgSet("-version"));
+#ifdef ENABLE_WALLET
+        const bool wallet_enabled{WalletModel::isWalletEnabled()};
```

Comment:
Whether this running instance has the wallet disabled at runtime or not is irrelevant to a new execution... Can fold the `#ifdef` deep into utilitydialog.cpp

Reply:
@luke-jr you mean that there's no need for `HelpMessageDialog` to have that third constructor argument at all? It could decide in it's implementation if wallet is compiled or not?

Reply:
Correct. The help message is about the build, not the current runtime.

Reply:
Ping @hebasto , @luke-jr has a point, or will your argue that run time (not compile time) changes should change help too?